src/app/clients/add/AddClientForm.tsx

```diff
@@ -0,0 +1,180 @@
+    FormText,
+} from "@/components/Form/formStyling";
+import fullNameCard from "@/app/clients/add/formSections/fullNameCard";
+import phoneNumberCard from "@/app/clients/add/formSections/phoneNumberCard";
+import addressCard from "@/app/clients/add/formSections/addressCard";
+import numberAdultsCard from "@/app/clients/add/formSections/numberAdultsCard";
+import numberChildrenCard from "@/app/clients/add/formSections/numberChildrenCard";
+import dietaryRequirementCard from "@/app/clients/add/formSections/dietaryRequirementCard";
+import feminineProductCard from "@/app/clients/add/formSections/feminineProductCard";
+import babyProductCard from "@/app/clients/add/formSections/babyProductCard";
+import petFoodCard from "@/app/clients/add/formSections/petFoodCard";
+import otherItemsCard from "@/app/clients/add/formSections/otherItemsCard";
+import deliveryInstructionsCard from "@/app/clients/add/formSections/deliveryInstructionsCard";
+import extraInformationCard from "@/app/clients/add/formSections/extraInformationCard";
+import {
+    ClientDatabaseRecord,
+    insertClient,
+    insertFamily,
+} from "@/app/clients/add/databaseFunctions";
+import attentionFlagCard from "@/app/clients/add/formSections/attentionFlagCard";
+import signpostingCallCard from "@/app/clients/add/formSections/signpostingCallCard";
+
+const initialFields: Fields = {
+    fullName: "",
+    phoneNumber: "",
+    addressLine1: "",
+    addressLine2: "",
+    addressTown: "",
+    addressCounty: "",
+    addressPostcode: "",
+    adults: [
+        { personType: "adult", quantity: 0 },
+        { personType: "male", quantity: 0 },
+        { personType: "female", quantity: 0 },
+    ],
+    numberChildren: 0,
+    children: [],
+    dietaryRequirements: {},
+    feminineProducts: {},
+    babyProducts: null,
+    nappySize: "",
+    petFood: {},
+    otherItems: {},
+    deliveryInstructions: "",
+    extraInformation: "",
+    attentionFlag: false,
+    signpostingCall: false,
+};
+
+const initialFormErrors: FormErrors = {
+    fullName: Error.initial,
+    phoneNumber: Error.none,
+    addressLine1: Error.initial,
+    addressPostcode: Error.initial,
+    adults: Error.initial,
+    numberChildren: Error.initial,
+    nappySize: Error.none,
+};
+
+const AddClientForm: React.FC = () => {
+    const router = useRouter();
+    const [fields, setFields] = useState<Fields>(initialFields);
+    const [formErrors, setFormErrors] = useState<FormErrors>(initialFormErrors);
+    const [submitError, setSubmitError] = useState<Error>(Error.none);
+
+    useEffect(() => {
+        if (fields.numberChildren <= fields.children.length) {
+            fieldSetter("children", fields.children.slice(0, fields.numberChildren));
+            return;
+        }
+
+        const extraChildren: Person[] = Array(fields.numberChildren - fields.children.length)
+            .fill(0)
+            // eslint-disable-next-line @typescript-eslint/no-unused-vars
+            .map((_item) => {
```

Comment:
Does this `// eslint-disable-next-line @typescript-eslint/no-unused-vars` still need to be here?

Reply:
For some reason it doesn't pass the linting test (both locally & on github) if it's not there - even though it has already been disabled.